src/merge.c

```diff
@@ -421,7 +421,7 @@ merge_iterator_create(platform_heap_id hid,
                      == ARRAY_SIZE(merge_itor->ordered_iterators),
                   "size mismatch");
 
-   merge_itor = TYPED_ZALLOC(hid, merge_itor);
+   merge_itor = TYPED_ZALLOC(PROCESS_PRIVATE_HEAP_ID, merge_itor);
```

Comment:
Renamed to this mnemonic (rather than `NULL_HEAP_ID`, which is what I had used during proto).

This is inline with a request from Rob for a better name.